detection-rules/impersonation_google_drive_file_share.yml

```
@@ -0,0 +1,137 @@
+                             "*shared with you*",
+                             "*invited you to review*",
+                             "*received a document*",
+                             "*shared a document*",
+                             "*shared this document*"
+               )
+        )
+      )
+      and (
+        strings.ilike(subject.subject,
+                      "*shared*",
+                      "*updated*",
+                      "*sign*",
+                      "*review*"
+        )
+        or strings.ilike(subject.subject, "*Docs*", "*Sheets*", "*Slides*")
+        or any(body.links, strings.icontains(.display_text, "open document"))
+        or subject.subject is null
+        or subject.subject == ""
+      )
+    )
+    or any([
+             "Contigo", // Spanish
+             "Avec vous", // French
+             "Mit Ihnen", // German
+             "Con te", // Italian
+             "Com você", // Portuguese
+             "Met u", // Dutch
+             "С вами", // Russian
+             "与你", // Chinese (Simplified)
+             "與您", // Chinese (Traditional)
+             "あなたと", // Japanese
+             "당신과", // Korean
+             "معك", // Arabic
+             "آپ کے ساتھ", // Urdu
+             "আপনার সাথে", // Bengali
+             "आपके साथ", // Hindi
+             "Sizinle", // Turkish // Azerbaijani
+             "Med dig", // Swedish
+             "Z tobą", // Polish
+             "З вами", // Ukrainian
+             "Önnel", // Hungarian
+             "Μαζί σας", // Greek
+             "איתך", // Hebrew
+             "กับคุณ", // Thai
+             "Với bạn", // Vietnamese
+             "Dengan Anda", // Indonesian // Malay
+             "Nawe", // Swahili
+             "Cu dumneavoastră", // Romanian
+             "S vámi", // Czech
+             "Med deg", // Norwegian
+             "S vami", // Slovak
+             "Med dig", // Danish
+             "Amb vostè", // Catalan
+             "Teiega", // Estonian
+             "S vama", // Serbian
+           ],
+           strings.icontains(subject.subject, .)
+    )
+  )
+  
+  // contains logic that impersonates Google
+  and (
+    any(ml.logo_detect(beta.message_screenshot()).brands,
+        strings.starts_with(.name, "Google")
+    )
+    or any(attachments,
+           .file_type in $file_types_images
+           and any(ml.logo_detect(.).brands, strings.starts_with(.name, "Google"))
+    )
+    or (
+      strings.icontains(body.current_thread.text,
+                        strings.concat("You have received this email because ",
+                                       sender.email.email,
+                                       " shared a document with you"
+                        )
+      )
+    )
+  )
+  and not all(body.links, .href_url.domain.root_domain in ("google.com"))
+  and sender.email.domain.root_domain not in $org_domains
+  and sender.email.domain.root_domain not in ("google.com")
```

```suggestion
  and sender.email.domain.root_domain not in ("google.com")
  and not (
    all(headers.references, strings.ends_with(., '@docs-share.google.com'))
    and headers.return_path.domain.domain == "doclist.bounces.google.com"
  )
```
Address [two](https://platform.sublime.security/messages/502d3eebed2e0dbb2a561bdc6f8c2fc93f1e71a11c44b233c399271336076da7?preview_id=019524d5-c2c2-7a78-8635-c59880409905) [observed](https://platform.sublime.security/rules/editor?canonical_id=502d3eebed2e0dbb2a561bdc6f8c2fc93f1e71a11c44b233c399271336076da7&message_id=019524d5-c2c2-7a78-8635-c59880409905) FPs